core/src/main/java/org/apache/iceberg/util/JsonUtil.java

```diff
@@ -65,6 +67,16 @@ public static Integer getIntOrNull(String property, JsonNode node) {
     return pNode.asInt();
   }
 
+  public static Long getLongOrNull(String property, JsonNode node) {
+    if (!node.has(property)) {
+      return null;
+    }
+    JsonNode pNode = node.get(property);
+    Preconditions.checkArgument(pNode != null && !pNode.isNull() && pNode.isIntegralNumber() &&
+        pNode.canConvertToLong(), "Cannot parse %s from non-string value: %s", property, pNode);
```

Comment:
just wondering whether the error message should mention that it can't parse it to a Long

Reply:
+1

Reply:
Yeah I would say `"Cannot parse %s to a long value: %s", property, pNode)`;